torch/distributed/distributed_c10d.py

```diff
@@ -816,7 +819,8 @@ class P2POp(object):
             ``torch.distributed.irecv``.
         tensor (Tensor): Tensor to send or receive.
         peer (int): Destination or source rank.
-        group (ProcessGroup, optional): The process group to work on.
+        group (ProcessGroup, optional): The process group to work on. If None,
```

Comment:
Hmm, not sure if I understand how this worked previously? If `group` passed in was always a process group, how could it be compared successfully against `group.WORLD` when it was an `object()` before this PR?

Reply:
I was surprised by this too and it is actually broken in some cases. In the latest update to the PR I've added a test called `test_pass_default_pg`, which passes with this PR but fails on master. I guess technically you can't really retrieve the default process group unless you use the private `_get_default_group` API, so I guess we were relying on that for this not to break. But I feel that is very fragile.